runtime/tests/jni/CMakeLists.txt

```diff
@@ -41,6 +41,10 @@ target_link_libraries(j9ben
 		j9thr
 		jvm
 )
+target_include_directories(j9ben
+	PRIVATE
+		"${j9vm_BINARY_DIR}/j9vm"
```

Comment:
Why does this need to use the binary dir as an include directory?  Is this now a requirement of anyone linking `j9vm_interface`?

Reply:
its needed to get `generated.h`. This patch also starts putting `generated.h` in the binary tree (previously it was generated in the source tree just for ease of porting)